settings/windowSettings.txt
[WindowSettings]

# This is the X coordinate where the message will begin to be displayed in the window. Increase this to move messages to the right. Decrease to move them to the left.
MESSAGE_X_POS = 60

# This is the X coordinate where the CENTER of your images should be displayed in the window. Increase this to move images to the right. Decrease to move them to the left.
IMAGE_X_POS = 22

# Width of the window in pixels. For the program to work properly, it is highly recommended to use a multiple of 4.
WINDOW_WIDTH = 400

# Height of the window in pixels. For the program to work properly, it is highly recommended to use a multiple of 4.
WINDOW_HEIGHT = 44

# Default window background color. Plain color words are accepted, or if you know hex, you can also type this in the format #123456 (include the # sign)
WINDOW_BG_COLOR = black

# This is the location of the background image. If no image is provided or the filepath is bad, the background color above will be used.
WINDOW_BG_IMAGE = imagefiles/background.png

# This is the X coordinate where the CENTER of your background image should be displayed in the window. Increase this to move it to the right. Decrease to move it to the left.
BACKGROUND_X_POS = 202

# This is the Y coordinate where the CENTER of your background image should be displayed in the window. Increase this to move it down. Decrease to move it up.
BACKGROUND_Y_POS = 24

# This is the delay in seconds between each move action. This simulates fluid movement of an object off the screen.
# Recommendation is not to mess with this, but you can try it for fun. Default value is .004
MOVE_ALL_ON_LINE_DELAY = .004

settings/messageSettings.txt
[MessageSettings]

# This is the duration in seconds of each message
MESSAGE_DURATION = 5

# This is the duration in seconds of the pause between the previous message disappearing and the next message appearing
MESSAGE_INTERMISSION = .5

# This is the default color of the text. Plain color words are accepted.
MESSAGE_COLOR = white

# Allows a modifier on the font, such as bold or italic.
MESSAGE_STYLE = bold

# This is the name of the font. STRONG PREFERENCE FOR MONOSPACED FONTS. If it can't find the font you list here, it will attempt to use a (bad) default.
MESSAGE_FONT_FACE = Courier New

# This is the maximum number of characters in a message before it will attempt to shrink the font. If you set this too high, your messages could trail outside of the window
MAX_LENGTH_FOR_NORMAL_FONT_SIZE = 16

# This is the normal font size of a message
NORMAL_FONT_SIZE = 26

# This is the horizontal gap in pixels between each character for a message that uses NORMAL_FONT_SIZE
NORMAL_FONT_SIZE_GAP = 20

# Read this carefully. This is the number of pixels to be subtracted from the usual gap between characters for a message that uses NORMAL_FONT_SIZE
# if the character is a space (e.g. " ").
# In other words, we don't want to waste a bunch of space in the window separating words. Set this to a higher number to move words closer together, or a lower number
# to move them farther apart.
NORMAL_FONT_SIZE_GAP_FOR_SPACES = 4

# This is the smaller font size used when a message exceeds the length of MAX_LENGTH_FOR_NORMAL_FONT_SIZE
SMALLER_FONT_SIZE = 22

# This is the horizontal gap in pixels between each character for a message that uses SMALLER_FONT_SIZE
SMALLER_FONT_SIZE_GAP = 16

# Read this carefully. This is the number of pixels to be subtracted from the usual gap between characters for a message that uses SMALLER_FONT_SIZE
# if the character is a space (e.g. " ").
# In other words, we don't want to waste a bunch of space in the window separating words. Set this to a higher number to move words closer together, or a lower number
# to move them farther apart.
SMALLER_FONT_SIZE_GAP_FOR_SPACES = 6

src/settings.py
import json

class Settings:
    def __init__(self):
        with open("settings.json") as f:
            self.data = json.loads(f.read())

        # Departure settings
        d = self.data['departure'][0]
        self.ENABLE_DEPARTING_BY_SLIDING_RIGHT = bool(d['ENABLE_DEPARTING_BY_SLIDING_RIGHT'] or False)
        self.ENABLE_DEPARTING_BY_SLIDING_LEFT = bool(d['ENABLE_DEPARTING_BY_SLIDING_LEFT'] or False)
        self.ENABLE_DEPARTING_BY_SLIDING_UP = bool(d['ENABLE_DEPARTING_BY_SLIDING_UP'] or False)
        self.ENABLE_DEPARTING_BY_SLIDING_DOWN = bool(d['ENABLE_DEPARTING_BY_SLIDING_DOWN'] or False)
        self.ENABLE_DEPARTING_BY_ALTERNATING_UP_AND_DOWN_WORKING_RIGHT_TO_LEFT = bool(d['ENABLE_DEPARTING_BY_ALTERNATING_UP_AND_DOWN_WORKING_RIGHT_TO_LEFT'] or False)
        self.ENABLE_DEPARTING_BY_ALTERNATING_UP_AND_DOWN_WORKING_LEFT_TO_RIGHT = bool(d['ENABLE_DEPARTING_BY_ALTERNATING_UP_AND_DOWN_WORKING_LEFT_TO_RIGHT'] or False)
        self.ENABLE_DEPARTING_BY_ALTERNATING_UP_AND_DOWN_IN_RANDOM_ORDER = bool(d['ENABLE_DEPARTING_BY_ALTERNATING_UP_AND_DOWN_IN_RANDOM_ORDER'] or False)
        self.ENABLE_DEPARTING_BY_COVERING_WITH_RECTANGLES = bool(d['ENABLE_DEPARTING_BY_COVERING_WITH_RECTANGLES'] or False)

        # Message settings
        m = self.data['message'][0]
        self.MESSAGE_DURATION = int(m['MESSAGE_DURATION'] or 5)
        self.MESSAGE_INTERMISSION = float(m['MESSAGE_INTERMISSION'] or .5)
        self.MESSAGE_COLOR = str(m['MESSAGE_COLOR'] or "white")
        self.MESSAGE_STYLE = str(m['MESSAGE_STYLE'] or "bold")
        self.MESSAGE_FONT_FACE = str(m['MESSAGE_FONT_FACE'] or "Courier New")
        self.MAX_LENGTH_FOR_NORMAL_FONT_SIZE = int(m['MAX_LENGTH_FOR_NORMAL_FONT_SIZE'] or 16)
        self.NORMAL_FONT_SIZE = int(m['NORMAL_FONT_SIZE'] or 26)
        self.NORMAL_FONT_SIZE_GAP = int(m['NORMAL_FONT_SIZE_GAP'] or 20)
        self.NORMAL_FONT_SIZE_GAP_FOR_SPACES = int(m['NORMAL_FONT_SIZE_GAP_FOR_SPACES'] or 4)
        self.SMALLER_FONT_SIZE = int(m['SMALLER_FONT_SIZE'] or 22)
        self.SMALLER_FONT_SIZE_GAP = int(m['SMALLER_FONT_SIZE_GAP'] or 16)
        self.SMALLER_FONT_SIZE_GAP_FOR_SPACES = int(m['SMALLER_FONT_SIZE_GAP_FOR_SPACES'] or 6)
        self.DEFAULT_IMAGE = str(m['DEFAULT_IMAGE'] or "imagefiles/stLogo28.png")

        # Window settings
        w = self.data['window'][0]
        self.MESSAGE_X_POS = int(w['MESSAGE_X_POS'] or 60)
        self.IMAGE_X_POS = int(w['IMAGE_X_POS'] or 21)
        self.WINDOW_WIDTH = int(w['WINDOW_WIDTH'] or 400)
        self.WINDOW_HEIGHT = int(w['WINDOW_HEIGHT'] or 44)
        self.WINDOW_BG_COLOR = str(w['WINDOW_BG_COLOR'] or "black")
        self.WINDOW_BG_IMAGE = str(w['WINDOW_BG_IMAGE'] or "imagefiles/background.png")
        self.BACKGROUND_X_POS = int(w['BACKGROUND_X_POS'] or 202)
        self.BACKGROUND_Y_POS = int(w['BACKGROUND_Y_POS'] or 24)
        self.MOVE_ALL_ON_LINE_DELAY = float(w['MOVE_ALL_ON_LINE_DELAY'] or .004)
